server/parse-route.js

```diff
@@ -0,0 +1,9 @@
+export default function parseRoute(hashRoute) {
+  if (hashRoute.startsWith('#')) {
+    hashRoute = hashRoute.replace('#', '');
+  }
+  const [, id] = hashRoute.split('=');
+  const [path, queryString] = hashRoute.split('?');
+  const params = new URLSearchParams(queryString);
+  return { path, params, id };
```

Comment:
You can leave the id in the url search params. it can be retrieved by key `params.get('productId')`. Also, this is not needed in your server code.